crypto/evp/mac_lib.c

```diff
@@ -189,3 +189,65 @@ int EVP_MAC_names_do_all(const EVP_MAC *mac,
 
     return 1;
 }
+
+unsigned char *EVP_Q_mac(OSSL_LIB_CTX *libctx, const char *name, const char *propq,
+                         const char *subalg, const OSSL_PARAM *params,
+                         const void *key, int keylen,
+                         const unsigned char *data, size_t datalen,
+                         unsigned char *out, size_t outsize, unsigned int *outlen)
```

Comment:
Similarly outlen should be a size_t?

Reply:
I am afraid that would make it impossible to implement the HMAC() macro in terms of EVP_Q_mac call.

Reply:
Same as above

Reply:
I agree that it should be size_t..
The function you are adding should match the parameter signature that we want - not the legacy function parameter.

HMAC is NOT a macro.. This means that { size_t len; XXX_Q(...&len); *outlen = len; } can be done.